packages/mcdoc/src/runtime/checker/index.ts

```diff
@@ -503,16 +503,34 @@ export function typeDefinition<T>(
 		for (const def of node.validDefinitions) {
 			if (!attached.has(node.node.originalNode)) {
 				options.attachTypeInfo(node.node.originalNode, def.typeDef)
-				handleAttributes(
-					def.typeDef.attributes,
-					options.context,
-					(handler, c) => {
-						const attacher = handler.attachString?.(c, options.context)
-						if (attacher) {
-							options.stringAttacher(node.node.originalNode, attacher)
+				const attributes = def.typeDef.attributes
+				handleAttributes(attributes, options.context, (handler, config) => {
+					const parser = handler.stringParser?.(config, options.context)
+					if (!parser) {
+						return
+					}
+					options.stringAttacher(node.node.originalNode, (node) => {
+						const src = new core.Source(node.value, node.valueMap)
+						const start = src.cursor
+						const child = parser(src, options.context)
+						if (!child) {
+							options.context.err.report(
+								localize(
+									'expected',
+									localize('mcdoc.runtime.checker.value'),
+								),
+								core.Range.create(start, src.skipRemaining()),
+							)
+							return
+						} else if (src.canRead()) {
+							options.context.err.report(
+								localize('mcdoc.runtime.checker.trailing'),
+								core.Range.create(src.cursor, src.skipRemaining()),
+							)
```

Comment:
error reproting should really be restricted to `reportErrors`, and they should be thrown when the nodes and definitions are first encountered for error condensing and filtering definitions.

Reply:
I disagree with that, see also my replies on #1245. If attributes want to affect the definition filtering they need to use `checkInferred` (like `id` does), The `stringParser` handler should assume that this is the only typedef possible for the node and runs after type info has been attached.

Reply:
Well, the point is, there is no reason to assume there is only one possible union type that has a string parser. Maybe this is true right now, but I still don't think this is the correct approach down the line. It might be fine for 4.0.